emulate/sender.py

```diff
@@ -2,16 +2,47 @@
 import websockets
 import redis
 import os
+from time import time
 
 
 redis_db = redis.Redis(
 	host=os.environ['REDIS_HOST'],
 	port=int(os.environ['REDIS_PORT'])
 )
+FADE_IN_TIME_MILLISECONDS = 1000
+FADE_OUT_TIME_MILLISECONDS = 1000
+
+
+def current_milliseconds():
+	return round(time() * 1000)
+
+
+def check_fadein_fadeout(dmx_values):
+	now_time = current_milliseconds()
+	code_start_time = int(redis_db.get('Code_start_time'))
+	code_end_time = int(redis_db.get('Code_end_time'))
+	
+	multiplier = 1.0
+	delta_from_start = now_time - code_start_time
+	delta_to_end = code_end_time - now_time
+	if delta_from_start < FADE_IN_TIME_MILLISECONDS: # fade in
+		multiplier = delta_from_start / FADE_IN_TIME_MILLISECONDS
+	elif delta_to_end < FADE_OUT_TIME_MILLISECONDS: # fade out
+		multiplier = delta_to_end / FADE_OUT_TIME_MILLISECONDS
+
+	dmx_values = [min(255, max(0, int(int(x) * (multiplier)))) for x in dmx_values.split(",")]
+	dmx_values = ",".join([str(e) for e in dmx_values])
+	return dmx_values
+	
 
 async def send_dmx_values(websocket, path):
 	while True:
 		dmx_values = redis_db.get('DMXvalues').decode('utf-8')
+		if dmx_values is None:
+			continue
+
+		dmx_values = check_fadein_fadeout(dmx_values)
```

Comment:
funkcja check_fadein_fadeout wskazuje ze powinna sprawdzać fade_in + fade_out a robi też updateowania wartości dmx!!!

Funkcja powinna robić tylko to co wskazuje nazwa!!!

lepiej zrobic funkcję get_fade_multiplier a wysyłanie + update wartosci w funkcji  send_dmx_values, lub w oddzielnej funkcji